01_financial-well-being/income_metrics_all.qmd

```diff
@@ -0,0 +1,676 @@
+
+```{r}
+acs2021clean %>% 
+  ggplot(aes(x = HHINCOME)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("HHIncome Variable Results") +
+  ylab("Density")
+
+acs2021clean %>% 
+  ggplot(aes(x = RENTGRS)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Gross Rent Variable Results") +
+  ylab("Density")
+
+acs2021clean %>% 
+  ggplot(aes(x = OWNCOST)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Homeowner cost Variable Results") +
+  ylab("Density")
+```
+
+```{r}
+
+acs2021clean  <- left_join(acs2021clean, puma_place, by=c("statefip","puma"))
+
+```
+
+Now have 3,886,784 observations Run anti_join to see how many cases on the left did not have a match on the right
+
+```{r}
+
+test  <- anti_join(acs_2021, puma_place, by=c("statefip","puma"))
+
+```
+
+1,656,456 obs from the microdata (makes sense since we limited to only PUMAs that are overlapping with Places of interest).
+
+Drop any observations with NA for afact (i.e. there is no place of interest overlapping this PUMA).
+
+```{r}
+
+acs2021clean <- acs2021clean %>% 
+  filter(!is.na(afact))
+
+```
+
+3,886,784 obs to 2,230,328 obs. Also filter out cases where AFACT is equal to zero.
+
+```{r}
+acs2021clean <- acs2021clean %>% 
+  filter(afact > 0)
+```
+
+Create a variable for the number of places per PUMA population per place (e.g., in unique statefip+place pairs).
+
+```{r}
+acs2021clean  <- acs2021clean  %>%
+  group_by(puma, statefip) %>%
+  mutate(places_per_PUMA = n_distinct(statefip, place))
+```
+
+Create a variable for the number of PUMAs per place.
+
+```{r}
+
+acs2021clean  <- acs2021clean %>%
+  group_by(place, statefip) %>%
+  mutate(PUMAs_per_place = n_distinct(statefip, puma))
+
+
+```
+
+Adjust weight to account for PUMA-to-county mapping (those where unique_types \> 1).Drop PUMA flag variable from PUMA_place.
+
+```{r}
+
+acs2021clean <- acs2021clean %>%
+  mutate(HHWT = HHWT*afact, 
+         HHINCOME = HHINCOME*ADJUST,
+         PERWT = PERWT*afact, 
+         RENTGRS = RENTGRS*ADJUST, 
+         OWNCOST = OWNCOST*ADJUST
+  ) %>% 
+  select(-puma_flag)
+
+```
+
+Remove group quarters.
+
+```{r}
+
+acs2021clean <- acs2021clean %>%
+  filter(GQ < 3)
+
+```
+
+Save as "microdata.csv"
+
+```{r}
+write_csv(acs2021clean, "data/intermediate/2021microdata.csv")
+
+```
+
+## Create income percentiles variable
+
+Create Income Metrics (non-subgroup)
+
+Objective: get pctl_20 pctl_50 and pctl_80 per unique state+place for year 2021 for the var HHINCOME.
+
+Aggregation should be weighted by HHWT
+
+Isolate data to count each household only once (PERNUM == 1 counts head of household only).
+
+```{r}
+acs2021income <- acs2021clean %>%
+  filter(PERNUM == 1)
+```
+
+Remove records where HHINCOME is missing - NA.
+
+```{r}
+acs2021income <- acs2021income %>%
+  filter(!is.na(HHINCOME))
```

Comment:
Filtering the missing values directly is risky. The universe is "Not group quarters, not vacant units." You shouldn't need to drop any missing values after the earlier filtering. Show that the values aren't missing and then delete this.